src/platform/packages/private/kbn-grid-layout/grid/grid_row/grid_row.tsx

```diff
@@ -98,54 +95,62 @@ export const GridRow = React.memo(({ rowIndex }: GridRowProps) => {
        * reasons (screen readers and focus management).
        */
       const gridLayoutSubscription = gridLayoutStateManager.gridLayout$.subscribe((gridLayout) => {
+        if (!gridLayout[rowIndex]) return;
         const newPanelIdsInOrder = getKeysInOrder(gridLayout[rowIndex].panels);
         if (panelIdsInOrder.join() !== newPanelIdsInOrder.join()) {
           setPanelIdsInOrder(newPanelIdsInOrder);
         }
       });
 
-      const columnCountSubscription = gridLayoutStateManager.runtimeSettings$
-        .pipe(
-          map(({ columnCount }) => columnCount),
-          distinctUntilChanged()
-        )
-        .subscribe((columnCount) => {
-          const rowRef = gridLayoutStateManager.rowRefs.current[rowIndex];
-          if (!rowRef) return;
-          rowRef.style.setProperty('--kbnGridRowColumnCount', `${columnCount}`);
-        });
-
       return () => {
         interactionStyleSubscription.unsubscribe();
         gridLayoutSubscription.unsubscribe();
         rowStateSubscription.unsubscribe();
-        columnCountSubscription.unsubscribe();
       };
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [rowIndex]
   );
 
+  const toggleIsCollapsed = useCallback(() => {
+    const newLayout = cloneDeep(gridLayoutStateManager.gridLayout$.value);
+    newLayout[rowIndex].isCollapsed = !newLayout[rowIndex].isCollapsed;
+    gridLayoutStateManager.gridLayout$.next(newLayout);
+  }, [rowIndex, gridLayoutStateManager.gridLayout$]);
+
+  useEffect(() => {
+    /**
+     * Set `aria-expanded` without passing the expanded state as a prop to `GridRowHeader` in order
+     * to prevent `GridRowHeader` from rerendering when this state changes
+     */
+    if (!collapseButtonRef.current) return;
+    collapseButtonRef.current.ariaExpanded = `${!isCollapsed}`;
+  }, [isCollapsed]);
+
   return (
-    <div css={styles.fullHeight} className="kbnGridRowContainer">
+    <div
+      css={styles.fullHeight}
+      className={classNames('kbnGridRowContainer', {
+        'kbnGridRowContainer--collapsed': isCollapsed,
+      })}
```

Comment:
Using a class name here is more performant so that things like the accordion's open/closed arrow can be controlled with raw CSS without a React re-render of `GridRowHeader`